File: POO/J/CarStore/src/business/Car.java
package business;
import java.awt.Color;
public class Car {
	public String model;
	public int maxSpeed;
	public Color color;
	public double zeroToHundred;
	public Engine engine;
	//
	public Car() {
		// Standard Constructor
	}
	//
	public Car(String model, int maxSpeed, Color color, double zeroToHundred, Engine engine) {
		super();
		this.model = model;
		this.maxSpeed = maxSpeed;
		this.color = color;
		this.zeroToHundred = zeroToHundred;
		this.engine = engine;
	}
	//
	@Override
	public String toString() {
		return String.format("Model: %s - Max Speed: %dkm/h - Color: %s - Zero to hundred: %.2f seconds - Type of Engine: %s - Engine's Potency: %d horses", model, maxSpeed, color, zeroToHundred, engine != null ? engine.type.toString() : "[No Engine]", engine != null ? engine.potency : 0);
	}
	//
	public String getModel() {
		return model;
	}
	public void setModel(String model) {
		this.model = model;
	}
	public int getMaxSpeed() {
		return maxSpeed;
	}
	public void setMaxSpeed(int maxSpeed) {
		this.maxSpeed = maxSpeed;
	}
	public Color getColor() {
		return color;
	}
	public void setColor(Color color) {
		this.color = color;
	}
	public double getZeroToHundred() {
		return zeroToHundred;
	}
	public void setZeroToHundred(double zeroToHundred) {
		this.zeroToHundred = zeroToHundred;
	}
}

File: POO/CS/CarStore/CarStore/CarStore/Business/Car.cs
using System;
using System.Collections.Generic;
using System.Drawing;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace CarStore.Business
{
    class Car
    {
        public Car()
        {
            // Standard Constructor
        }
        public Car(String Model, int MaxSpeed, Color Color, double ZeroToHundred, Engine Engine)
        {
            this.Model = Model;
            this.MaxSpeed = MaxSpeed;
            this.Color = Color;
            this.ZeroToHundred = ZeroToHundred;
            this.Engine = Engine;
        }
        public override string ToString()
        {
            return String.Format("Model: {0} - Max Speed: {1}km/h - Color: {2} - Zero to hundred: {3:F2} seconds - Type of Engine: {4} - Engine's Potency: {5} horses", Model, MaxSpeed, Color, ZeroToHundred, ((Engine != null) ? Engine.Type.ToString() : "[No Engine]"), Engine != null ? Engine.Potency : 0);
        }

        public string Model { get; set; }
        public int MaxSpeed { get; set; }
        public Color Color { get; set; }
        public double ZeroToHundred { get; set; }
        public Engine Engine { get; set; }
    }
}
